lucene/core/src/java/org/apache/lucene/search/SortField.java

```diff
@@ -392,4 +491,31 @@ public SortField rewrite(IndexSearcher searcher) throws IOException {
   public boolean needsScores() {
     return type == Type.SCORE;
   }
+
+  /**
+   * Returns an {@link IndexSorter} used for sorting index segments by this SortField.
+   *
+   * If the SortField cannot be used for index sorting (for example, if it uses scores or
+   * other query-dependent values) then this method should return {@code null}
+   *
+   * SortFields that implement this method should also implement a constructor that
+   * takes a {@link DataInput} for deserialization, to match the {@link IndexSorter#serialize(DataOutput)}
+   * method on the returned IndexSorter
+   */
```

Comment:
make it `lucene.internal`?

Reply:
I'm not sure this API is particularly internal? If people want to implement their own SortFields and make them indexable they will need to implement this method, which makes it public IMO.

We should probably take `@lucene.experimental` off `getFieldComparator()` as well, it's been there since 2004...

Reply:
I suggested it because I was seeing your change as a cleanup to the way that we support configuring index sorts. experimental would work for me too, what I like is to signal users that we feel free to make breaking changes to these APIs.